app/Foliage/HackageSecurity.hs

```diff
@@ -44,8 +47,23 @@ createKeys base = do
   replicateM_ 3 $ createKey' KeyTypeEd25519 >>= writeKeyWithId (base </> "mirrors")
 
 writeKeyWithId :: FilePath -> Some Key -> IO ()
-writeKeyWithId base k =
-  writeKey (base </> keyIdString (someKeyId k) <.> "json") k
+writeKeyWithId base k = do
+  let keyId' = keyIdString $ someKeyId k
+  let publicKey' = somePublicKey k
+
+  json <- toJSON publicKey'
+  case json of
+    JSObject obj -> do
+      case lookup "keyval" obj of
+        Just (JSObject keyValObj) ->
+          case lookup "public" keyValObj of
+            Just (JSString publicKeyVal) -> do
+              printf "Generated key %s with public key: %s\n" keyId' publicKeyVal
+            _ -> pure ()
+        _ -> pure ()
+    _ -> pure ()
```

Comment:
The triple `_ -> pure ()` is indeed ugly, but I can't find a better Haskell way of writing it :/

Reply:
Could use `ViewPatterns`:
```
case json of 
  JSObject (lookup "keyval" -> Just (JSObject (lookup "public" -> Just (JSString publicKeyVal)))) -> ...
```
Or you could do something with lenses, but probably not worth it.

Reply:
Why decoding the json though? You have the public key already.

Reply:
But I wanted the `base64` representation of the public key, and I should admit that the dumbest way I quickly found to pretty print it …

Reply:
I admit the hackage-security API can feel over-engineere sometimes. You see I had to define a bunch of utilities of my own. We just need to add a couple more :P

```
import Crypto.Sign.Ed25519 (unPublicKey)
import Data.ByteString.Base16 qualified as Base16

writeKeyWithId :: FilePath -> Some Key -> IO ()
writeKeyWithId base k = do
  let keyId' = keyIdString $ someKeyId k
  let publicKey' = somePublicKey k
  print $ Base16.encode $ exportSomePublicKey publicKey'

  writeKey (base </> keyId' <.> "json") k

exportSomePublicKey :: Some PublicKey -> ByteString
exportSomePublicKey (Some k) = exportPublicKey k

exportPublicKey :: PublicKey a -> ByteString
exportPublicKey (PublicKeyEd25519 pub) = unPublicKey pub
```